worktree.go

```diff
@@ -897,6 +900,130 @@ func (w *Worktree) Submodule(name string) (*Submodule, error) {
+}
+
+// resolveSSHRelativeURL resolves relative URLs for SSH protocol
+func (w *Worktree) resolveSSHRelativeURL(baseEndpoint *transport.Endpoint, relativeURL, originURL string) (string, error) {
+	// For SSH URLs like git@github.com:user/repo.git
+	if baseEndpoint.Host == "" || baseEndpoint.Path == "" {
+		return "", fmt.Errorf("invalid SSH URL format for origin: %s", originURL)
+	}
+
+	basePath := strings.TrimSuffix(baseEndpoint.Path, ".git")
+	pathParts := strings.Split(basePath, "/")
+
+	relativeParts := strings.Split(relativeURL, "/")
+	for _, part := range relativeParts {
+		if part == ".." && len(pathParts) > 0 {
+			pathParts = pathParts[:len(pathParts)-1]
+		} else if part != "." && part != "" {
+			pathParts = append(pathParts, part)
+		}
+	}
+
+	newPath := strings.Join(pathParts, "/")
+
+	if baseEndpoint.User != "" {
+		return fmt.Sprintf("%s@%s:%s", baseEndpoint.User, baseEndpoint.Host, newPath), nil
+	}
+	return fmt.Sprintf("%s:%s", baseEndpoint.Host, newPath), nil
+}
+
+// resolveFileRelativeURL resolves relative URLs for file:// protocol
+func (w *Worktree) resolveFileRelativeURL(baseEndpoint *transport.Endpoint, relativeURL string) (string, error) {
+	// For file:// URLs like file:///path/to/repo.git
+	if !strings.HasSuffix(relativeURL, ".git") {
+		return "", fmt.Errorf("file:// protocol requires .git suffix, got: %s", relativeURL)
+	}
+
+	baseDir := path.Dir(baseEndpoint.Path)
+	resolvedPath := path.Join(baseDir, relativeURL)
+
+	return fmt.Sprintf("file://%s", resolvedPath), nil
+}
+
+// getOriginRemoteURL extracts the origin remote URL for reuse
+func (w *Worktree) getOriginRemoteURL() (string, error) {
+	remotes, err := w.r.Remotes()
+	if err != nil {
+		return "", err
+	}
+
+	var originRemote *Remote
+	for _, remote := range remotes {
+		if remote.c.Name == "origin" {
+			originRemote = remote
+			break
+		}
+	}
+
+	if originRemote == nil {
+		return "", fmt.Errorf("no origin remote found")
+	}
+
+	if len(originRemote.c.URLs) == 0 {
+		return "", fmt.Errorf("origin remote has no URLs configured")
+	}
+
+	return originRemote.c.URLs[0], nil
+}
+
+// resolveRelativeSubmoduleURL resolves relative URLs using a pre-fetched origin URL
+func (w *Worktree) resolveRelativeSubmoduleURL(relativeURL, originURL string) (string, error) {
+	if !isRelativeURL(relativeURL) {
+		return "", fmt.Errorf("URL is not relative: %s", relativeURL)
+	}
+
+	baseEndpoint, err := transport.NewEndpoint(originURL)
+	if err != nil {
+		return "", err
+	}
+
+	// Handle different protocols
+	switch baseEndpoint.Protocol {
+	case "https", "http":
+		return w.resolveHTTPRelativeURL(originURL, relativeURL)
+	case "ssh":
+		return w.resolveSSHRelativeURL(baseEndpoint, relativeURL, originURL)
+	case "file":
+		return w.resolveFileRelativeURL(baseEndpoint, relativeURL)
+	default:
+		// For other protocols, we don't support relative URL resolution
+		return "", fmt.Errorf("relative URL resolution not supported for protocol: %s", baseEndpoint.Protocol)
+	}
+}
```

Comment:
What about:

```suggestion
// resolveModuleURL resolves relative URLs based on the originURL.
// The moduleURL is returned verbatim when originURL is empty or moduleURL is an absolute URL.
func resolveModuleURL(originURL, moduleURL string) (string, error) {
	if originURL == "" {
		return moduleURL, nil
	}

	if !strings.HasPrefix(moduleURL, "../") && !strings.HasPrefix(moduleURL, "./") {
		return moduleURL, nil
	}

	if !giturl.MatchesScheme(originURL) && giturl.MatchesScpLike(originURL) {
		user, host, portStr, path := giturl.FindScpLikeComponents(originURL)

		path = filepath.Join(path, moduleURL)
		if portStr != "" {
			portStr += ":"
		}

		return fmt.Sprintf("%s@%s:%s%s", user, host, portStr, path), nil
	}

	base, err := url.Parse(originURL)
	if err != nil {
		return "", err
	}

	base.Path = filepath.Join(base.Path, moduleURL)

	return base.String(), nil
}
```

The only use case that needs to be handled differently is SCP-like URLs, everything else seems to work fine with `url.Parse`.